### PR TITLE
gh-124761: add socket.SO_REUSEPORT_LB

### PR DESCRIPTION
**Pull request by graingert**
* Issue: gh-124761

### CODE REVIEW

**Comment by graingert** [comment]
this is only half of the work as the flag needs to be used when reuse_port=True is passed to various apis

File: Misc/NEWS.d/next/Library/2024-10-04-09-56-45.gh-issue-124761.N4pSD6.rst
Comment (picnixz) [inline Misc/NEWS.d/next/Library/2024-10-04-09-56-45.gh-issue-124761.N4pSD6.rst]:
```suggestion
Add :data:`~socket.SO_REUSEPORT_LB` constant to :mod:`socket` for FreeBSD.
```

File: Doc/library/socket.rst
Comment (picnixz) [inline Doc/library/socket.rst:679]: Should `SO_REUSEPORT_LB` be preferred on FreeBSD instead of `SO_REUSEPORT`?

File: Doc/library/socket.rst
Comment (picnixz) [inline Doc/library/socket.rst]: Could we specify the minimal FreeBSD version?

**Review by picnixz — COMMENTED** [review]
In the PR for `SO_ORIGINAL_DST` that I just reviewed, I didn't really feel the need to document it since we don't really document every constant (and it was better to have the constant included without going through a round of documentation review).

Here, since it's FreeBSD-only I think it's nice to document it (`SO_ORIGINAL_DST` is exposed assuming some headers, which would probably make its docs a bit messy I think). Since there's no conditional header, I appreciate the additional documentation.

Note: I haven't thought about it but we could have actually added some `.. versionadded` to the `SO_*` entries though for `SO_ORIGINAL_DST`.

---

I'd also like @erlend-aasland's review on this one if possible.

File: Doc/library/socket.rst
Comment (graingert) [reply Doc/library/socket.rst]:
```suggestion
  .. availability:: FreeBSD >= 12.0
```

https://www.freebsd.org/releases/12.0R/relnotes/

Comment (picnixz) [reply Doc/library/socket.rst]: I think we can use `FreeBSD >= 12.0` since we already write "Linux >= ...".

File: Doc/library/socket.rst
Comment (graingert) [reply Doc/library/socket.rst:679]: Yeah, the original ticket is to add the flag then use it in socket.create_connection, I thought I'd split it into two PRs for easier review

Comment (picnixz) [reply Doc/library/socket.rst:679]: I'm not sure if it could he helpful but we could mention that it should be preferrable to use it instead of `SO_REUSEPORT` then (I'm not sure whether we should recommend what to use in the docs).

Comment (graingert) [reply Doc/library/socket.rst:679]: I think we should leave out the recommendation for now

**Review by erlend-aasland — APPROVED** [review]

File: Doc/library/socket.rst
Comment (picnixz) [inline Doc/library/socket.rst]:
```suggestion
  .. versionadded:: next
```

We can now use this new directive!

**Review by picnixz — APPROVED** [review]

**Comment by graingert** [comment]
@picnixz can I get a merge?

**Comment by picnixz** [comment]
Er... I'm not a core dev! we should ask @erlend-aasland ~~or maybe you give me commit rights for this one~~

**Comment by picnixz** [comment]
(I'm still not a core dev, I still can't merge this one :()

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**
